### PR TITLE
[extractor/wordpress] Add generic playlist embed extractor

### PR DESCRIPTION
**Pull request by coletdjnz**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

</details>

ADD DESCRIPTION HERE

Fixes https://github.com/yt-dlp/yt-dlp/issues/4955

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [X] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [X] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [X] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [X] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [X] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### CODE REVIEW

**Comment by coletdjnz** [comment]
Marked as draft as core yt-dlp seems to have issues with us yielding multiple playlists (unless I've screwed something up)

cc @pukkandan

**Comment by pukkandan** [comment]
The nested playlist issue is fixed

File: yt_dlp/extractor/wordpress.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        # class should always be "wp-playlist-script"
+        # See: https://core.trac.wordpress.org/browser/trunk/src/wp-includes/media.php#L2930
+        for i, j in enumerate(get_elements_by_class('wp-playlist-script', webpage)):
+            playlist_json = self._parse_json(j, self._generic_id(url), fatal=False, ignore_extra=True, errnote='') or {}
+            if not playlist_json:
+                continue
+            entries = []
+            for track in playlist_json.get('tracks') or []:
+                if not isinstance(track, dict):
+                    continue
```
Comment (pukkandan) [inline yt_dlp/extractor/wordpress.py:59]:
```suggestion
            for track in traverse_obj(playlist_json, ('tracks', ...), expected_type=dict):
```

I would have written the loop as
```py
entries = [{
    ...
} for track in traverse_obj(playlist_json, ('tracks', ...), expected_type=dict)]
```

File: yt_dlp/extractor/wordpress.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                    'title': track.get('title'),
+                    'url': track.get('src'),
+                    'thumbnail': traverse_obj(track, ('thumb', 'src')),
+                    'album': traverse_obj(track, ('meta', 'album')),
+                    'artist': traverse_obj(track, ('meta', 'artist')),
+                    'genre': traverse_obj(track, ('meta', 'genre')),
+                    'duration': parse_duration(traverse_obj(track, ('meta', 'length_formatted'))),
+                    'description': track.get('description'),
+                    'height': int_or_none(traverse_obj(track, ('dimensions', 'original', 'height'))),
+                    'width': int_or_none(traverse_obj(track, ('dimensions', 'original', 'width'))),
```
Comment (pukkandan) [inline yt_dlp/extractor/wordpress.py:71]: Due to the recent improvements in `traverse_obj`, this is now possible.
```py
                    ...,
                    **traverse_obj(track, {
                        'url': 'src',
                        'thumbnail': ('thumb', 'src'),
                        'album': ('meta', 'album'),
                        'artist': ('meta', 'artist'),
                        'genre': ('meta', 'genre'),
                        'description': 'description',  
                    })
```

Not saying we should use it (I'm undecided on whether this is good idea or not), just letting you know the possibility exists now

**Comment by pukkandan** [comment]
If you decide to address #4994 in this PR, ping me. I'll want to review again

**Comment by coletdjnz** [comment]
> If you decide to address #4994 in this PR, ping me. I'll want to review again

I'll do another PR for that

**Pull request merged**